packages/core/data-transfer/src/strapi/remote/handlers/pull.ts

```diff
@@ -175,6 +175,7 @@ export const createPullController = handlerControllerFactory<Partial<PullHandler
         error: null,
         id,
       });
+      batch = [];
```

Comment:
This is just enforcing something that is necessary rather than allowing room for potentially bugs.